mnemosyne/repair.py

```
@@ -0,0 +1,1028 @@
+def _verify_report_gate(
+    manifest: _RepairManifest,
+    db_path: Path,
+    bank_name: str,
+    *,
+    conn: sqlite3.Connection | None = None,
+    bound_identity: dict[str, int] | None = None,
+) -> None:
+    """Fail closed unless this exact bank, file, and schema still match."""
+
+    if manifest.bank_name != bank_name:
+        raise RepairError("Repair report bank does not match the selected database")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)